_posts/2024-XX-XX-quarkus-kiota.adoc

```diff
@@ -0,0 +1,165 @@
+- To ease the fruition of the Kiota CLI from standard Java projects, we implemented a link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#maven-plugin[Maven plugin].
+- To make frictionless the usage of Kiota on different and opinionated technological stacks, we ship alternative core libraries to let the user easily swap out from Microsoft's defaults (OkHttp + Gson): link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#serialization-jackson[Jackson Serde], link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#http-vertx[Vert.X Http], link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#http-jdk[JDK Http].
+- To increase compatibility with alternative Java runtimes, we completely removed the usage of reflection from all of the internals. Thus, Kiota-generated SDKs are automatically, and with zero configuration, able to be compiled and run on link:https://www.graalvm.org/latest/reference-manual/native-image/[GraalVM native-image].
+- Feeling the pain of un-maintained and various levels of maturity of URI Templates implementations (internally used to compose URLs), we rolled out a link:https://github.com/std-uritemplate/std-uritemplate?tab=readme-ov-file#motivation[dependency-free unified library] available for all of the languages Kiota supports.
+
+Along with additional bug fixes and improvements, we finally achieved a sweet spot where Kiota can be easily integrated and leveraged by mainstream and mature software projects. link:https://github.com/Apicurio/apicurio-registry[Apicurio Registry stands out among others] shipping (and extensively leveraging it in the tests) the generated link:https://github.com/Apicurio/apicurio-registry/tree/main/java-sdk[Java SDK] along with the link:https://github.com/Apicurio/apicurio-registry/tree/main/python-sdk[Python SDK] and the link:https://github.com/Apicurio/apicurio-registry/tree/main/go-sdk[Go SDK].
+
+=== Quarkus Meet Kiota
+
+Now that all the dots have been connected, the final and crucial step is a seamless integration with supersonic, subatomic applications leveraging Quarkus!
+This is the motivation behind the birth of the new ``quarkus-kiota`` extension.
+The project is in its early stages, and we encourage you to try it out and provide feedback.
+
+The codebase lives in the link:https://github.com/quarkiverse/quarkus-kiota[Quarkiverse], the project is listed in the link:https://quarkus.io/extensions/io.quarkiverse.kiota/quarkus-kiota/[extensions] and the link:https://docs.quarkiverse.io/quarkus-kiota/dev/index.html[docs] are available at the usual location.
+
+Get started by adding this extension:
+
+----
+quarkus ext add io.quarkiverse.kiota:quarkus-kiota
+----
+
+Since it's a code generator extension, you will need to add the `generate-code` goal to the `quarkus-maven-plugin` `executions` section:
```

Comment:
```suggestion
Since it's a code generator extension, you will want to check the `generate-code` goal is present in the `quarkus-maven-plugin` `executions` section. Default Quarkus generated projects have it but custom or older projects might not:
```